diff --git a/arch/arm/dts/k3-am642-sk.dts b/arch/arm/dts/k3-am642-sk.dts
index d4721e3a4e4..b8e02f1393e 100644
--- a/arch/arm/dts/k3-am642-sk.dts
+++ b/arch/arm/dts/k3-am642-sk.dts
@@ -19,6 +19,7 @@
 
 	chosen {
 		stdout-path = &main_uart0;
+		bootargs = "console=ttyS2,115200n8 earlycon=ns16550a,mmio32,0x02800000 security=apparmor systemd.unified_cgroup_hierarchy=0";
 	};
 
 	aliases {
@@ -367,7 +368,8 @@
 
 	main_ecap0_pins_default: main-ecap0-default-pins {
 		pinctrl-single,pins = <
-			AM64X_IOPAD(0x0270, PIN_INPUT, 0) /* (D18) ECAP0_IN_APWM_OUT */
+			AM64X_IOPAD(0x0270, PIN_OUTPUT, 1) /* (D18) ECAP0_IN_APWM_OUT */
+			AM64X_IOPAD(0x0274, PIN_OUTPUT, 1) /* (A19) EXT_REFCLK1 */ 
 		>;
 	};
 
@@ -386,6 +388,23 @@
 		>;
 	};
 
+
+	mygpio0_pins_default: mygpio0-pins-default {
+			pinctrl-single,pins = <
+					AM64X_IOPAD(0x00a4, PIN_OUTPUT, 7) /* (N17) GPMC0_DIR.GPIO0_40 */
+			>;
+	};
+
+	rpi_spi0_pins_default: rpi-spi0-pins-default {
+		pinctrl-single,pins = <
+			AM64X_IOPAD(0x0210, PIN_OUTPUT, 0) /* D13 SPI0_CLK.SPI0_CLK */
+			AM64X_IOPAD(0x0208, PIN_OUTPUT, 0) /* D12 SPI0_CS0.SPI0_CS0 */
+			AM64X_IOPAD(0x020C, PIN_OUTPUT, 0) /* D12 SPI0_CS1.SPI0_CS1 */
+			AM64X_IOPAD(0x0214, PIN_INPUT, 0)  /* A13 SPI0_D0.SPI0_D0 */
+			AM64X_IOPAD(0x0218, PIN_OUTPUT, 0) /* A14 SPI0_D1.SPI0_D1 */
+		>;
+	};
+
 	main_com8_ls_en_pins_default: main-com8-ls-en-default-pins {
 		pinctrl-single,pins = <
 			AM64X_IOPAD(0x00fc, PIN_OUTPUT, 7) /* (U7) PRG1_PRU0_GPO17.GPIO0_62 */
@@ -410,6 +429,15 @@
 	pinctrl-0 = <&main_uart0_pins_default>;
 	current-speed = <115200>;
 };
+ 
+&main_gpio0 {
+       pinctrl-names = "default";
+       pinctrl-0 = <&mygpio0_pins_default>;
+
+       gpio0-40 {
+               gpios = <40 GPIO_ACTIVE_HIGH>;
+       };
+};
 
 &main_uart1 {
 	/* main_uart1 is reserved for firmware usage */
@@ -460,6 +488,24 @@
 	};
 };
 
+&main_spi0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&rpi_spi0_pins_default>;
+	ti,spi-num-cs = <2>;
+	spidev@0 {
+			spi-max-frequency = <1000000>;
+			reg = <0>;
+			compatible = "spidev";
+	};
+	spidev@1 {
+			spi-max-frequency = <1000000>;
+			reg = <1>;
+			compatible = "spidev";
+	};
+};
+
+
 /* mcu_gpio0 and mcu_gpio_intr are reserved for mcu firmware usage */
 &mcu_gpio0 {
 	status = "reserved";
@@ -740,8 +786,8 @@
 		pinctrl-single,pins = <
 			/* pps [cpts genf1] in22 -> out37 [cpts hw8_push] */
 			TS_OFFSET(37, 22)
-			/* pps [cpts genf1] in22 -> out26 [SYNC1_OUT pin] */
-			TS_OFFSET(26, 22)
+			/* pps [cpts genf1] in22 -> out24 [SYNC1_OUT pin] */
+			TS_OFFSET(24, 22)
 			>;
 	};
 };
